File: Udemy/Sec_12_PictureInPicture/app/src/main/java/com/example/sec_12_pictureinpicture/MainActivity.java
package com.example.sec_12_pictureinpicture;

import androidx.appcompat.app.AppCompatActivity;

import android.content.res.Configuration;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    public void goPIP(View view){
//         Set in manifest
//        android:resizeableActivity="true"
//        android:supportsPictureInPicture="true"
//        android:configChanges="screenSize|smallestScreenSize|screenLayout|orientation"
        enterPictureInPictureMode();
    }

    @Override
    public void onPictureInPictureModeChanged(boolean isInPictureInPictureMode, Configuration newConfig) {
        super.onPictureInPictureModeChanged(isInPictureInPictureMode, newConfig);

        TextView textView = findViewById(R.id.textView);
        Button pipButton = findViewById(R.id.button);

        if(isInPictureInPictureMode){
            // Going into PiP
            pipButton.setVisibility(View.INVISIBLE);
            getSupportActionBar().hide();
            textView.setText("$10,004");
        }else{
            // Going out of Pip
            pipButton.setVisibility(View.VISIBLE);
            getSupportActionBar().show();
            textView.setText("Hello");
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


}

File: Udemy/Sec_9_KotlinAndroidFun/app/src/main/java/com/example/sec_9_kotlinandroidfun/MainActivity.kt
package com.example.sec_9_kotlinandroidfun

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var count = 0

    fun reset(view : View){
        count = 0
        textView.text = count.toString()
    }

    fun plusOne(view : View){
        count++
        textView.text = count.toString()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var textView = findViewById<TextView>(R.id.textView)
        textView.text = "Hello Kotlin : Count 0"
    }
}
